### PR TITLE
Add `float` option

### PR DESCRIPTION
**Pull request by aichholzer**
I found myself wanting to make the box float on the right of my screen. This simple update takes care of it.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -140,6 +140,14 @@ Space around the box.
 
 Accepts a number or an object with any of the `top`, `right`, `bottom`, `left` properties. When a number is specified, the left/right margin is 3 times the top/bottom to make it look nice.
 
+##### float
+
+Type: `string`<br>
+Values: `right` `center` `left`<br/>
+Default: `right`
```
Comment (sindresorhus) [inline readme.md]: This contradicts the actual code, which defines `left` as default.

File: readme.md [hunk 2]
```diff
@@ -140,6 +140,14 @@ Space around the box.
 
 Accepts a number or an object with any of the `top`, `right`, `bottom`, `left` properties. When a number is specified, the left/right margin is 3 times the top/bottom to make it look nice.
 
+##### float
+
+Type: `string`<br>
+Values: `right` `center` `left`<br/>
```
Comment (sindresorhus) [inline readme.md]: Remove the `/`

File: index.js [hunk 3]
```diff
@@ -115,7 +116,25 @@ module.exports = function (text, opts) {
...
+	var currentColumns = process.stdout.columns;
+	switch (opts.float) {
+		case 'center':
+			padWidth = (currentColumns / 2) - (contentWidth / 2);
+			marginLeft = repeating(PAD, padWidth);
+			break;
+
+		case 'right':
+			padWidth = currentColumns - contentWidth - 2;
+			padWidth = padWidth < 0 ? 0 : padWidth;
```
Comment (sindresorhus) [inline index.js]:
```js
padWidth = Math.max(currentColumns - contentWidth - 2, 0);
```

File: index.js [hunk 4]
```diff
@@ -115,7 +116,25 @@ module.exports = function (text, opts) {
...
 	var contentWidth = widestLine(text) + padding.left + padding.right;
 	var paddingLeft = repeating(PAD, padding.left);
-	var marginLeft = repeating(PAD, margin.left);
+
+	var marginLeft;
+	var padWidth;
+	var currentColumns = process.stdout.columns;
+	switch (opts.float) {
+		case 'center':
+			padWidth = (currentColumns / 2) - (contentWidth / 2);
```
Comment (sindresorhus) [inline index.js]:
```js
padWidth = (currentColumns - contentWidth) / 2;
```

**Comment by sindresorhus** [comment]
Needs a test ;)

File: index.js [hunk 5]
```diff
@@ -115,7 +116,25 @@ module.exports = function (text, opts) {
...
+
+		case 'right':
+			padWidth = currentColumns - contentWidth - 2;
+			padWidth = padWidth < 0 ? 0 : padWidth;
+			marginLeft = repeating(PAD, padWidth);
+			break;
+
+		default: marginLeft = repeating(PAD, margin.left);
+			break;
+	}
```
Comment (SamVerschueren) [inline index.js]: I'd rather like to see this logic being extracted in a separate function?

File: index.js [hunk 6]
```diff
@@ -115,7 +116,25 @@ module.exports = function (text, opts) {
...
+			marginLeft = repeating(PAD, padWidth);
+			break;
+
+		case 'right':
+			padWidth = currentColumns - contentWidth - 2;
+			padWidth = padWidth < 0 ? 0 : padWidth;
+			marginLeft = repeating(PAD, padWidth);
+			break;
+
+		default: marginLeft = repeating(PAD, margin.left);
```
Comment (SamVerschueren) [inline index.js]: Add a newline after `default:`

**Comment by marcjansen** [comment]
Wouldn't it make more sense if this were an option for thw cli? Just thinking out loudly.

File: readme.md (hunk 1 above)
Comment (aichholzer) [inline readme.md]: Yep, typo on my side.

**Comment by sindresorhus** [comment]
@marcjansen I think both makes sense. I could see myself using this programatically too.

**Comment by sindresorhus** [comment]
Thank you @aichholzer :)

Would you be interested in adding support for the option to the [CLI](https://github.com/sindresorhus/boxen-cli)?

**Comment by aichholzer** [comment]
@sindresorhus 
👍🏼 You are welcome, glad I help.
Sure, no problem. I will fork it and make a pull request.

**Pull request merged**